xclim/subset.py

```diff
@@ -125,11 +133,188 @@ def func_checker(*args, **kwargs):
+
+    Parameters
+    ----------
+    da : Union[xarray.DataArray, xarray.Dataset]
+      Input data.
+    shape : Union[str, Path, geometry.GeometryCollection]
+      Path to a single-layer vector file, or a shapely GeometryCollection object.
+    start_date : Optional[str]
+      Start date of the subset.
+      Date string format -- can be year ("%Y"), year-month ("%Y-%m") or year-month-day("%Y-%m-%d").
+      Defaults to first day of input data-array.
+    end_date : Optional[str]
+      End date of the subset.
+      Date string format -- can be year ("%Y"), year-month ("%Y-%m") or year-month-day("%Y-%m-%d").
+      Defaults to last day of input data-array.
+    da_crs : Optional[Union[int, dict, str]]
+      CRS of the xarray.DataArray or xarray.Dataset provided. Default: dict(epsg=4326).
+    shape_crs : Optional[Union[int, dict, str]]
+      CRS of the geometries provided. If passing GeometryCollections as shapes, CRS must be explicitly stated.
+    use_all_features : bool
+      Use either the first found feature geometry or the union of all found features. Default: False.
+    start_yr : int
+      Deprecated
+        First year of the subset. Defaults to first year of input data-array.
+    end_yr : int
+      Deprecated
+        Last year of the subset. Defaults to last year of input data-array.
+
+    Returns
+    -------
+    Union[xarray.DataArray, xarray.Dataset]
+      Subsetted xarray.DataArray or xarray.Dataset
+
+    Warnings
+    --------
+    This functions relies on the rioxarray library and requires xarray Datasets and DataArrays that have been read with
+     the `rioxarray` library imported. Attempting to use this function with pure xarray objects will raise exceptions.
+
+    Examples
+    --------
+    >>> from xclim import subset
+    >>> import rioxarray
+    >>> import xarray as xr
+    >>> ds = xarray.open_dataset('pr.day.nc')
+    Subset lat lon and years
+    >>> prSub = subset.subset_shape(ds.pr, shape="/path/to/polygon.shp", start_yr='1990', end_yr='1999')
+    Subset data array lat, lon and single year
+    >>> prSub = subset.subset_shape(ds.pr, shape="/path/to/polygon.shp", start_yr='1990', end_yr='1990')
+    Subset data array single year keep entire lon, lat grid
+    >>> prSub = subset.subset_bbox(ds.pr, start_yr='1990', end_yr='1990') # one year only entire grid
+    Subset multiple variables in a single dataset
+    >>> ds = xarray.open_mfdataset(['pr.day.nc','tas.day.nc'])
+    >>> dsSub = subset.subset_bbox(ds, shape="/path/to/polygon.shp", start_yr='1990', end_yr='1999')
+     # Subset with year-month precision - Example subset 1990-03-01 to 1999-08-31 inclusively
+    >>> prSub = subset.subset_time(ds.pr, shape="/path/to/polygon.shp", start_date='1990-03', end_date='1999-08')
+    # Subset with specific start_dates and end_dates
+    >>> prSub = \
+            subset.subset_time(ds.pr, shape="/path/to/polygon.shp", start_date='1990-03-13', end_date='1990-08-17')
+    """
+    try:
```

Comment:
Add a few comments to explain what's going on.

Reply:
Done.